File: client/src/rule/ChessAction.ts
import Chess from "./Chess";
import ChessPos from "./ChessPos";
import ChessHost from "./chess_host";

export default class ChessAction {
  chessHost: ChessHost;

  chessType: any;

  fromPos: ChessPos;

  toPos: ChessPos;

  eatenChess: Chess;
}

File: server/src/main/java/io/github/hulang1024/chess/play/ChessAction.java
package io.github.hulang1024.chess.play;

import io.github.hulang1024.chess.play.rule.Chess;
import io.github.hulang1024.chess.play.rule.ChessEnum;
import io.github.hulang1024.chess.play.rule.ChessHost;
import io.github.hulang1024.chess.play.rule.ChessPos;
import lombok.Data;

@Data
public class ChessAction {
    private ChessHost chessHost;
    private ChessEnum chessType;
    private ChessPos fromPos;
    private ChessPos toPos;
    private Chess eatenChess;
}